lib/distro.py

```diff
@@ -57,10 +57,10 @@ def build_packages(self, packages):
         """
         self.package_builder.initialize()
         for package in packages:
+            # The package unlock is done inside package_builder.build()
             package.lock()
             package.download_files(recurse=False)
             self.package_builder.build(package)
-            package.unlock()
```

Comment:
It is simpler to understand the locking mechanism if lock() and unlock() are in the same level. I suggest moving the source code packaging to outside package_builder.build().